File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/mariadb/DateTimeTypes.scala
package dev.typr.foundations.docs.mariadb

import dev.typr.foundations.{MariaType, MariaTypes}
import java.time.{LocalDate, LocalDateTime, LocalTime, Year}

@SuppressWarnings(Array("unused"))
object DateTimeTypes:
  //start
  val dateType: MariaType[LocalDate] = MariaTypes.date
  val timeType: MariaType[LocalTime] = MariaTypes.time
  val datetimeType: MariaType[LocalDateTime] = MariaTypes.datetime
  val yearType: MariaType[Year] = MariaTypes.year

  // With fractional seconds precision
  val timeFsp: MariaType[LocalTime] = MariaTypes.time(6)
  val dtFsp: MariaType[LocalDateTime] = MariaTypes.datetime(3)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/mariadb/DateTimeTypes.java
package dev.typr.foundations.docs.mariadb;

import dev.typr.foundations.MariaType;
import dev.typr.foundations.MariaTypes;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.Year;

@SuppressWarnings("unused")
public class DateTimeTypes {
  // start
  MariaType<LocalDate> dateType = MariaTypes.date;
  MariaType<LocalTime> timeType = MariaTypes.time;
  MariaType<LocalDateTime> datetimeType = MariaTypes.datetime;
  MariaType<Year> yearType = MariaTypes.year;

  // With fractional seconds precision
  MariaType<LocalTime> timeFsp = MariaTypes.time(6);
  MariaType<LocalDateTime> dtFsp = MariaTypes.datetime(3);
  // stop
}
